backend/src/main/java/com/redhat/cloud/notifications/processors/email/EmailAggregator.java

```diff
@@ -60,42 +57,38 @@
-                                        .onItem().transform(Set::copyOf)
-                                        // Now we want to determine who will actually receive the aggregation email.
-                                        .onItem().transformToUni(endpoints ->
-                                                // All users who subscribed to the current application and subscription type combination are recipients candidates.
-                                                subscribers.onItem().transformToUni(users ->
-                                                        /*
-                                                         * The actual recipients list may differ from the candidates depending on the endpoint properties and the action settings.
-                                                         * The target endpoints properties and the action settings will determine whether or not each candidate will actually receive an email.
-                                                         */
-                                                        recipientResolver.recipientUsers(
-                                                                aggregationKey.getAccountId(),
-                                                                Stream.concat(
-                                                                        endpoints
-                                                                                .stream()
-                                                                                .map(EndpointRecipientSettings::new),
-                                                                        ActionRecipientSettings.fromAction(action).stream()
-                                                                ).collect(Collectors.toSet()),
-                                                                users
-                                                        )
-                                                )
-                                        )
-                                        /*
-                                         * We now have the final recipients list.
-                                         * Let's populate with synchronized side-effects (invoke) the Map that will be returned by the method.
-                                         */
-                                        .onItem().invoke(users -> {
-                                            users.forEach(user -> {
-                                                // It's aggregation time!
-                                                fillUsers(aggregationKey, user, aggregated, aggregation);
-                                            });
-                                        });
+                    // We need its event type to determine the target endpoints.
+                    String eventType = getEventType(aggregation);
```

Comment:
This is still not quite right, but it's what we had before (and similar to what we had before this PR).
We might need to collect all event types so that it's better.